src/Compilers/Core/Portable/InternalUtilities/StringOrdinalComparer.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+
+namespace Roslyn.Utilities
+{
+    /// <summary>
+    /// Compares string based upon their ordinal equality.
```

Comment:
because it handles both cases sensitive and insensitive scenarios and thus more complex. 
It also defaults to String.GetHashCode that is basically the same since CLR 1.0 for compat reasons. 

We have cases in binding that do a lot of hashtable lookups. It makes sense to have comparer with fewer indirections and more suitable for our scenarios.

Will add a comment.